JniBridge.bee.cs

```diff
@@ -0,0 +1,397 @@
+        "::java::util::HashSet",
+        "::java::util::Map_Entry",
+        "::java::util::NoSuchElementException",
+        "::java::util::Scanner",
+        "::java::util::zip::ZipFile",
+        "::javax::net::ssl::X509TrustManager",
+        "::javax::net::ssl::TrustManagerFactory",
+        "::java::security::KeyStore",
+        "::com::google::android::gms::ads::identifier::AdvertisingIdClient",
+        "::com::google::android::gms::common::GooglePlayServicesAvailabilityException",
+        "::com::google::android::gms::common::GooglePlayServicesNotAvailableException",
+    };
+
+    private static readonly string[] kMacOSClasses = new[]
+    {
+        "::java::lang::System",
+        "::java::lang::UnsupportedOperationException",
+    };
+
+    static void Main()
+    {
+    	using var _ = new BuildProgramContext();
+        var codegen = CodeGen.Release;
+
+        var jdk = SetupJava();
+        var sdk = SetupAndroidSdk();
+
+        var apiGenerator = SetupApiGenerator(jdk);
+        var jnibridgeJar = SetupJniBridgeJar(jdk);
+        var generatedFilesAndroid = SetupSourceGeneration(jdk, apiGenerator, jnibridgeJar, GetAndroidSourceGenerationParams(sdk));
+        var generatedFilesMacOS = SetupSourceGeneration(jdk, apiGenerator, jnibridgeJar, GetMacOSSourceGenerationParams(jdk));
+
+        var androidZip = new ZipArchiveContents();
+        var versionFile = VersionControl.SetupWriteRevisionInfoFile(new NPath("artifacts").Combine("build.txt"));
+        var androidToolchains = GetAndroidToolchains();
+        foreach (var toolchain in androidToolchains)
+        {
+            var androidConfig = new NativeProgramConfiguration(codegen, toolchain, false);
+            var np = SetupJniBridgeStaticLib(generatedFilesAndroid, androidConfig, GetAndroidStaticLibParams(toolchain), androidZip);
+        }
+        var headers = SetupJniBridgeHeaders(generatedFilesAndroid, "android");
+        foreach (var header in headers)
+        {
+            androidZip.AddFileToArchive(header, new NPath("include").Combine(header.FileName));
+        }
+        androidZip.AddFileToArchive(jnibridgeJar);
+        androidZip.AddFileToArchive(versionFile);
+
+        var osxToolchain = ToolChain.Store.Mac().Sdk_10_13().x64();
+        var osxConfig = new NativeProgramConfiguration(codegen, osxToolchain, false);
+        var osxStaticLib = SetupJniBridgeStaticLib(generatedFilesMacOS, osxConfig, GetMacOSStaticLibParams(osxToolchain, jdk));
+        SetupTestProgramOsx(osxToolchain, osxStaticLib, codegen, generatedFilesMacOS, jdk);
+
+        var androidZipPath = "build/builds.zip";
+        ZipTool.SetupPack(androidZipPath, androidZip);
+        Backend.Current.AddAliasDependency("build:android:zip", androidZipPath);
+    }
+
+    static Jdk SetupJava()
+    {
+        var jdk = Jdk.UserDefault;
+        if (jdk != null)
+            return jdk;
+        var openJdk = StevedoreArtifact.UnityInternal(HostPlatform.Pick(
```

Comment:
On more question, since we use Bee now, thus that mean any external party which clones this branch, will no longer be able to build jnibridge, since Bee downloads stuff from Stevedore, and non Unity people don't have access to it?

Reply:
Yeah, it seems so.